eva/udfs/object_detector.py

```diff
@@ -0,0 +1,238 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List
+
+import numpy as np
+import pandas as pd
+
+from eva.models.catalog.frame_info import FrameInfo
+from eva.models.catalog.properties import ColorSpace
+from eva.udfs.abstract.pytorch_abstract_udf import PytorchAbstractClassifierUDF
+import sys
+import cv2
+import importlib
+import warnings
+warnings.filterwarnings("ignore", category=UserWarning)
+
+try:
+    from torch import Tensor
+except ImportError as e:
+    raise ImportError(
+        f"Failed to import with error {e}, \
+        please try `pip install torch`"
+    )
+
+try:
+    import torchvision
+except ImportError as e:
+    raise ImportError(
+        f"Failed to import with error {e}, \
+        please try `pip install torch`"
+    )
+
+
+class ObjectDetector(PytorchAbstractClassifierUDF):
+    """
+    Arguments:
+        threshold (float): Threshold for classifier confidence score
+    """
+
+    def __init__(self, model_name):
+        module = importlib.import_module("torchvision.models.detection")
+        self._torch_model = getattr(module, model_name)
+        super().__init__()
```

Comment:
This code is essentially a copy paste of fasterrcnn example except for this minor modification. 

This is all we need to bring in any model supported by `torchvision.models.detection`